src/peripheral/dwt.rs

```diff
@@ -400,25 +411,50 @@ impl Comparator {
                     (_, EmitOption::PC) => return Err(DwtError::InvalidFunction),
                 };
 
-                self.function.modify(|mut r| {
-                    r.set_function(function);
-                    r.set_emitrange(emit_range);
-
+                (
+                    function,
+                    emit_range,
                     // don't compare data value
-                    r.set_datavmatch(false);
-
+                    false,
                     // don't compare cycle counter value
                     // NOTE: only needed for comparator 0, but is SBZP.
-                    r.set_cycmatch(false);
+                    false,
+                    settings.address,
+                    settings.mask,
+                )
+            }
+            ComparatorFunction::CycleCount(settings) => {
+                (
+                    // emit a Debug Watchpoint event, either halting execution or
+                    // firing a `DebugMonitor` exception
+                    0b0111,
+                    // emit_range is N/A for cycle count compare
+                    false,
+                    // don't compare data
+                    false,
+                    // compare cyccnt
+                    true,
+                    settings.compare,
+                    settings.mask,
+                )
```

Comment:
I'm not a fan of the tuples. It'll become difficult to read when `ComparatorFunction` is further extended. Return a structure instead.

Reply:
Fair point. I changed to a tuple to consolidate where we're updating the register values and cut down on duplicate code, but if we want a structure for clarity, might as well just have the duplicate locations, so I switched back to that.

Reply:
An alternative is to implement some `fn apply` on that struct.